src/tools/pub/pubspec.md

```diff
@@ -398,6 +402,30 @@ the pattern is considered relative to the package's root directory.
   Support for the `false_secrets` field was added in Dart 2.15.
 {{site.alert.end}}
 
+### Screenshots
+
+Packages may showcase their widgets or other visual elements using screenshots.
+
+A package can include up to 10 screenshots. These are specified in the pubspec as a list under the `screenshots` field. 
+Each screenshot is specified by a `description` and a `path`. The `description` is a string of at most 160 characters describing the screenshot. 
+For example:
+
+
+```yaml
+screenshots:
+  - description: 'description of screenshot'
+    path: path/to/image/in/package/500x500.webp
+  - description: 'description of second screenshot'
+    path: path/to/image/in/package.png
+```
+
+Supported file types include : `png`, `jpg`, `gif`, `webp`. Both static and animated files are supported. The screenshots shown on the pub.dev site
+will all be `webp` images generated from the original image. 
+
+The first screenshot in the list will be used for a thumbnail. Thumbnails will always be static. In case the original image is animated the first frame will be used for the thumbnail. We generate quadratic thumbnails to be shown on the pub.dev site. Hence, for the first screenshot, we recommend a 1:1 aspect ratio for best fitting. If the ratio is not 1:1, the image will be resized keeping the original aspect ratio. 
+
+The maximum allowed image file size is 4MB.
```

Comment:
Should we add a reminder that the files also will be included in each download of the package so it is preferred to keep them small?

Reply:
Hmm, I think that is quite clear, but maybe I am biased :-)

Reply:
Not saying it's unclear - just a friendly reminder about the implications of including large screenshots.

Reply:
4MB seems pretty large for a screenshot of a widget. I think it wouldn't hurt to have a small mention about this, since I imagine many users don't consider how a package download includes everything.

Do these images contribute to the 100 MB cap after gzip compression? That may make this mention even more relevant.

Reply:
Added a note.